### ISSUE TITLE
str2var raises a blocking script error when passed an invalid string, should maybe return null

### ISSUE THREAD

**Issue by cjmxp** [godotengine/godot#11457]
Because GDScript does not have Try... Except... Causes the following code to handle errors

func _ready():
	var db = str2var("")
	print("is error?",db)

**Comment by akien-mga** [comment]
I don't understand. What's the problem?

**Comment by marcosbitetti** [comment]
Oh. str2var raises an error with empty string.

**Comment by cjmxp** [comment]
@akien-mga 
There is no output here. The program will host

**Comment by Dillybob92** [comment]
Lol, I wish GDScript would add in try catch error handling, but I don't think that's **why** this error exists. Simply str2var should just return null or -1 if no input, right is the bug?

But then again,  the godot developer can also just do 

```
	var db = ""
	if db.length() != 0:
		db = str2var("")
		print("is error?",db)
```

so...

**Comment by NathanWarden** [comment]
@Dillybob92 You're code is fine. The string has a length of zero therefore it will never enter the if statement.

If you put this instead, it'll enter the else portion of the code.

```
	var db = ""
	if db.length() != 0:
		db = str2var("")
		print("is error?",db)
	else:
		print("db is 0 length as expected!")
```

**Comment by NathanWarden** [comment]
Yep, I think you meant to put "db.length() == 0"  :)

**Comment by Dillybob92** [comment]
Oops, wasn't clear in my post my bad.

I mean if the godot developer checked for length before using str2var, the error will never happen. So is this an issue with str2var inherently, or an issue surrounding the methodology of using GDScript?

**Comment by akien-mga** [comment]
The more you post, the farther we go from understanding what this is all about :D

What's the problem with the original code? What does `db` contain? To quote the issue template that was completely ignored:
- **Issue description:** What happened, and what was expected.
- **Steps to reproduce:**

**Comment by NathanWarden** [comment]
@Dillybob92 I should've read more of the above post and I probably would've seen what you were communicating more clearly ;)

@akien-mga Here's the problem quite simply:

```
func _ready():
    var db = str2var(some_var) # If some_var is an empty string then an exception is thrown here that there's no way to detect.
    print("is error?",db) # This line is never reached
```

So, exception handling would be the ultimate solution overall, but I think in the meantime I agree that it could simply return null.

Returning -1 wouldn't be a good idea since you don't know whether it failed or if your value in the string is actually -1

**Comment by akien-mga** [comment]
Alright, now I get it :)

It's actually not due to passing an empty string but any kind of invalid string (i.e. that does not match what `var2str` would produce for a given Variant type): https://github.com/godotengine/godot/blob/8632408dbd50223256a140940a5f02fb040a79da/modules/gdscript/gd_functions.cpp#L656-L681

Instead of raising a blocking error, it could maybe return `Variant()` (null) and just raise a non-blocking error in the debugger.

**Comment by cjmxp** [comment]
@akien-mga  The decoding error should return null

**Comment by SeleckyErik** [comment]
Taking up this one

**Comment by akien-mga** [comment]
Fixed by #22934.
